tests/unit/io-writer.spec.js

```diff
@@ -0,0 +1,154 @@
+import _ from 'lodash'
+import fs from 'fs'
+import test from 'tape'
+import Excel from 'exceljs'
+import tempWrite from 'temp-write'
+import Writer from '../../src/io/writer'
+
+const readXlsx = function readXlsx (filePath) {
+  const values = []
+  const workbook = new Excel.Workbook()
+  return workbook.xlsx.readFile(filePath)
+    .then(() => {
+      const worksheet = workbook.getWorksheet(1)
+      let headerLen = null
+      worksheet.eachRow((row) => {
+        let rowValues = row.values
+        rowValues.shift()
+
+        if (headerLen)
+          // older exceljs reader version trims last value
+          // if it is an empty string, so we will pad row
+          // to header length with empty string
+          rowValues = [
+            ...rowValues,
+            ...Array(headerLen - rowValues.length).fill(''),
+          ]
+        else
+          headerLen = rowValues.length
+
+        rowValues = _.map(rowValues,
+          item => (_.isObject(item) || _.isNil(item) ? '' : item))
+        values.push(rowValues)
+      })
+      return values
+    })
+}
+
+/* eslint-disable no-unused-vars */
+test(`Writer
+  should throw an error when outputFile is not specified`, (t) => {
+  const expectedOutput = 'Error: OutputFile was not specified'
+
+  try {
+    const writer = new Writer({
+      exportFormat: 'csv',
+    })
+    t.fail('Should throw an error')
+  } catch (e) {
+    t.equal(e.toString(), expectedOutput, 'Should throw an error')
+    t.end()
+  }
+})
+
+test(`Writer
+  should throw an error when incorrect export format is specified`, (t) => {
+  const expectedOutput = 'Error: Unsupported file type: unknown,' +
+    ' alowed formats are xlsx,csv'
+
+  try {
+    const writer = new Writer({
+      outputFile: 'out.csv',
+      exportFormat: 'unknown',
+    })
+
+    t.fail('Should throw an error')
+  } catch (e) {
+    t.equal(e.toString(), expectedOutput, 'Should throw an error')
+    t.end()
+  }
+})
+
+test(`Writer
+  should throw an error when incorrect encoding is specified`, (t) => {
+  const expectedOutput = 'Error: Encoding does not exist: unknown'
+
+  try {
+    const writer = new Writer({
+      outputFile: 'out.csv',
+      exportFormat: 'csv',
+      encoding: 'unknown',
+    })
+
+    t.fail('Should throw an error')
+  } catch (e) {
+    t.equal(e.toString(), expectedOutput, 'Should throw an error')
+    t.end()
+  }
+})
+
+test(`Writer
+  should return an error when writing fails`, (t) => {
+  const filePath = tempWrite.sync()
+  const expectedOutput = 'TypeError: must start with number, ' +
```

Comment:
For me this equals to `'TypeError: First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object.'`, please double check this is correct.